utils/payload/limiter_setup.go

```diff
@@ -54,14 +54,25 @@ func SetupAdaptiveLimiter(ctx context.Context, g *errgroup.Group) AdaptiveLimite
 			case <-time.After(statsFrequency):
 				limiterStats := limiter.Stats()
 
-				stats.Default.NewStat("adaptive_payload_limiter_state", stats.GaugeType).Gauge(limiterStats.State)
-				stats.Default.NewStat("adaptive_payload_limiter_threshold_factor", stats.GaugeType).Gauge(limiterStats.ThresholdFactor)
+				stats.Default.NewStat(
+					"adaptive_payload_limiter_state",
+					stats.GaugeType,
+				).Gauge(float64(limiterStats.State))
```

Comment:
The underlying type is an `int` really
```suggestion
				).Gauge(int(limiterStats.State))
```